src/main/kotlin/io/emeraldpay/dshackle/upstream/near/NearLowerBoundBlockDetector.kt

```diff
@@ -0,0 +1,25 @@
+package io.emeraldpay.dshackle.upstream.near
+
+import io.emeraldpay.dshackle.Chain
+import io.emeraldpay.dshackle.Global
+import io.emeraldpay.dshackle.upstream.LowerBoundBlockDetector
+import io.emeraldpay.dshackle.upstream.Upstream
+import io.emeraldpay.dshackle.upstream.rpcclient.JsonRpcRequest
+import reactor.core.publisher.Mono
+
+class NearLowerBoundBlockDetector(
+    chain: Chain,
+    val upstream: Upstream,
+) : LowerBoundBlockDetector(chain, upstream) {
+
+    override fun lowerBlockDetect(): Mono<LowerBlockData> {
+        return upstream.getIngressReader().read(JsonRpcRequest("status", listOf())).map {
+            val resp = Global.objectMapper.readValue(it.getResult(), NearStatus::class.java)
+            LowerBlockData(resp.syncInfo.earliestHeight, null, resp.syncInfo.earliestBlockTime.toEpochMilli())
+        }
+    }
+
+    override fun periodRequest(): Long {
+        return 120
```

Comment:
Maybe decrease period if near has super cool method with `earliestHeight`? Like in solana

Reply:
Ok! It was just copypaste from starknet. 
Which value is preferable?

Reply:
I suggest 3 (minutes), like in solana